### DISCUSSION TITLE
joining after startup

### DISCUSSION THREAD

**Discussion by Arminhosseini** [Koenkk/zigbee2mqtt#7289] [Devices]
Hi,
I have been working on this project for about 2 years but I have a continuous problem.
when I'm restarting this code or whenever the raspberry restart that makes this code restart all of my joined devices withdraw the network or at least I think so, and I have to join them all.
Is there any solution to this problem?
or am I doing anything wrong?

**Comment by Koenkk** [comment]
Could you provide the herdsman debug logging when starting and losing your network?

See https://www.zigbee2mqtt.io/information/debug.html#zigbee-herdsman-debug-logging on how to enable the herdsman debug logging. Note that this is only logged to STDOUT and not to log files.

**Comment by Arminhosseini** [comment]
Hi,
thank you for answering.
I attached the 
[afterunplug.txt](https://github.com/Koenkk/zigbee2mqtt/files/6413731/afterunplug.txt)
output after unplug and re-running the code.
that one of my devices with address 0x000b57fffed268cc loses its connection and gives me this error when I sending commands to that.

zigbee2mqtt:debug 2021-05-03 08:34:38: Error: Data request failed with error: 'No network route' (205)
    at ZStackAdapter.<anonymous> (/home/root/zigbee2mqtt/node_modules/zigbee-herdsman/dist/adapter/z-stack/adapter/zStackAdapter.js:578:27)
    at Generator.next (<anonymous>)
    at fulfilled (/home/root/zigbee2mqtt/node_modules/zigbee-herdsman/dist/adapter/z-stack/adapter/zStackAdapter.js:5:58)
zigbee2mqtt:info  2021-05-03 08:34:38: MQTT publish: topic 'application/0x00124b00043ebe29/node/bridge/log', payload '{"type":"zigbee_publish_error","message":"Publish 'set' 'state' to '0x000b57fffed268cc' failed: 'Error: Data request failed with error: 'No network route' (205)'","meta":{"friendly_name":"0x000b57fffed268cc"}}'

**Comment by Arminhosseini** [comment]
and this is my logs when I'm unplugging the whole power system including devices.
as you see most of my devices don't work anymore.
[afterunplugpower.txt](https://github.com/Koenkk/zigbee2mqtt/files/6414814/afterunplugpower.txt)

**Comment by Koenkk** [comment]
You are using a very old z2m version (1.8.0, release on 15 december 2019). Update to the latest version first and make sure your adapter uses the latest firmware.

**Reply by Arminhosseini** [reply]
are you sure this will solve my problem?
cause I need quickest and the most confident solution for now.
